src/main/java/org/opentripplanner/model/plan/ScheduledTransitLeg.java

```diff
@@ -64,7 +64,7 @@ public class ScheduledTransitLeg implements TransitLeg {
   private final Float accessibilityScore;
   private List<FareProductUse> fareProducts = List.of();
 
-  ScheduledTransitLeg(ScheduledTransitLegBuilder<?> builder) {
+  protected ScheduledTransitLeg(ScheduledTransitLegBuilder<?> builder) {
```

Comment:
I am encountering a bug that prevents trips from being planned when consolidated stops are involved.
The bug was not present prior to the last merge with the `dev-2.x` branch (the merge that includes migration to Java 21).

I think what is happening is that the call to `ScheduledTransitLegBuilder(ScheduledTransitLeg original)` that creates this `builder` variable is not picking up the time zone `zoneId` from `original` when trip patterns are modified with a shared stop.

Multi-agency trip searches that don't involve consolidated stops appear to work fine.

Here is the call using ATL feeds and the consolidated stop data below:

Query:
http://localhost:8001/?module=planner&fromPlace=33.92228087152904%2C-84.17012214660646&toPlace=33.78442642789061%2C-84.34118270874025&time=6%3A20am&date=11-14-2023&mode=TRANSIT%2CWALK&arriveBy=false&wheelchair=false&showIntermediateStops=true&locale=en&baseLayer=OSM%20Standard%20Tiles

`consolidated-stops.csv` (entire file):
```csv
stop_group_id,feed_id,stop_id,is_primary
600026,GwinnettCountyTransit,1064,0
600026,MARTA,70296,1
600027,GwinnettCountyTransit,1054,0
600027,MARTA,70300,1
```

Stack trace excerpt:
```
10:02:14.155 ERROR [grizzly-7]  (PlannerResource.java:109) System error
java.lang.NullPointerException: zone
	at java.base/java.util.Objects.requireNonNull(Objects.java:259)
	at java.base/java.time.ZonedDateTime.ofLocal(ZonedDateTime.java:370)
	at java.base/java.time.ZonedDateTime.of(ZonedDateTime.java:295)
	at java.base/java.time.ZonedDateTime.of(ZonedDateTime.java:267)
	at org.opentripplanner.framework.time.ServiceDateUtils.asStartOfService(ServiceDateUtils.java:45)
	at org.opentripplanner.framework.time.ServiceDateUtils.toZonedDateTime(ServiceDateUtils.java:68)
	at org.opentripplanner.model.plan.ScheduledTransitLeg.getIntermediateStops(ScheduledTransitLeg.java:265)
	at org.opentripplanner.api.mapping.LegMapper.mapLeg(LegMapper.java:130)
	at org.opentripplanner.api.mapping.LegMapper.mapLegs(LegMapper.java:51)
	at org.opentripplanner.api.mapping.ItineraryMapper.mapItinerary(ItineraryMapper.java:47)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.AbstractList$RandomAccessSpliterator.forEachRemaining(AbstractList.java:722)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682)
	at org.opentripplanner.api.mapping.ItineraryMapper.mapItineraries(ItineraryMapper.java:25)
	at org.opentripplanner.api.mapping.TripPlanMapper.mapTripPlan(TripPlanMapper.java:27)
	at org.opentripplanner.api.resource.PlannerResource.plan(PlannerResource.java:75)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        ...
```

Reply:
The relevant change to `ScheduledTransitLegBuilder` is in this commit https://github.com/ibi-group/OpenTripPlanner/commit/ef5f54cb3cbdc2399355e9c439ae0a44c04b894a from #5455.

Reply:
Are you sure that you have up to date transi data? That error usually happens when all of the trips in the past.

Reply:
Yes, I used the same data to review previous iterations of this PR.

Reply:
@binh-dam-ibigroup You're right, there was a bug in accessibility score code. I fixed it here: https://github.com/opentripplanner/OpenTripPlanner/pull/5511

There is another (unmerged) bug fix for accessible routes here: https://github.com/opentripplanner/OpenTripPlanner/pull/5516

Neither have anything to do with this PRs.

I have now merged dev-2.x and this is ready for review and testing again.